### ISSUE TITLE
TSV<->CSV conversion failing 

### ISSUE THREAD

**Issue by davebx** [galaxyproject/galaxy#9341]
```
Traceback (most recent call last):
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 44, in <module>
    main()
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 22, in main
    convert_to_csv(input_fname, output_fname)
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 40, in convert_to_csv
    writer.writerow(line.strip().split('\t'))
TypeError: a bytes-like object is required, not 'str'
```

**Comment by jdavcs** [comment]
It's a Py2>Py3 issue. In Python2 you need `open(file, 'rb')`, in Python3 `open(file, newline='')`. Neither one will run under the other.

**Comment by davebx** [comment]
That makes me think of an ugly workaround of checking whether it's running under py2 or py3 before opening the file. I don't like it, but it would _work_.

**Comment by jdavcs** [comment]
I was thinking the same. Can't think of any other way other than `sys.version_info`.

**Comment by davebx** [comment]
I'm going to try something with `six` first, but I may have to fall back on that.

**Comment by jdavcs** [comment]
Same in `tabular.py`; possibly in `job_metrics/collectl/processes.py`

**Comment by davebx** [comment]
```
{epidermis}dave@gaius:~/github-repos/galaxy $ find lib -iname \*.py -exec grep -Hn 'open(' {} \; | grep "'.b'" | wc -l
109
```

I'm not sure all of these will break, but it seems opening something `'_b'` can cause problems in one or the other.

**Comment by jdavcs** [comment]
'b' is still valid mode for the `open` function in Py3, so I guess it shouldn't be an issue.

**Comment by davebx** [comment]
Yeah, I _think_ it's only an issue if one attempts to do stringy stuff on a file handle opened that way?

**Comment by jennaj** [comment]
Exact same py2>py3 error for tabular_csv.py (and probably other similar) converters. I wasn't sure if all were captured already. Ok to put here or do we want distinct tickets for each converter with the same syntax issue?

Test histories: 
* ORG (py3): https://usegalaxy.org/u/jen/h/test-tabularcsvpy
* EU (py2): https://usegalaxy.eu/u/jenj/h/test-tabularcsvpy

**Comment by jdavcs** [comment]
@jennaj - thanks, yes, that would be the place for `tabular_csv`. If there are others, I would open a separate issue (maybe one covering all?). 
(The links you included are not accessible: "History is not accessible by user")

**Comment by jennaj** [comment]
@ic4f -- thanks. Ok -- agree, we probably need a ticket for all that links into some sort of batch search that listing all the tools with the issue -- many include the same problematic code chunk. For some reason, I thought this was identified already but I guess not. @mvdbeek or @natefoo -- could one of you confirm this wouldn't be a duplicate? I didn't find any recent updates for text manipulation/converter tools tossing this same error, included in the base release, with a search -- just this one.

&& oh rats, I've noticed share links not persisting the "first time" created. It has only been a few days since that started happening, but this helps to confirm it is reproducible (again), so thanks for pointing that out! I toggled the share URL on/off, should stick now. So ... I will also open a ticket for that -- was a problem at least once before (at both ORG and EU) -- cannot recall the reason/fix/release but will look for the prior PR and link it in, could be a regression issue.

thanks!
